### ISSUE TITLE
how to fine tune yolov11 by adding 2  specific classes for object dectection?

### ISSUE THREAD

**Issue by FrankTseng168** [ultralytics/ultralytics#17897]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

how to fine tune yolov11 by adding 2  specific classes for object dectection?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @FrankTseng168, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a custom training ❓ Question on how to fine-tune YOLOv11 for two specific classes, please provide additional details to assist us in helping you more effectively:
- Are you building a custom dataset? If so, share insights on dataset format, labels, and structure. 
- Are you experiencing issues during training, or are you seeking guidance on best practices?

Please verify that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which include critical considerations for custom training. If this becomes a 🐛 Bug Report during implementation, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv11 can be trained in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to assist you faster 😊. An Ultralytics engineer will review your question and provide further assistance as soon as possible.

**Comment by Y-T-G** [comment]
You can't add classes to a model. You have to train on all of the classes that you want to detect.

**Comment by FrankTseng168** [comment]
Dear Mohammed Yasin,
Thanks for your email.

Mohammed Yasin ***@***.***> 於 2024年12月1日 週日 下午2:53寫道：

**Comment by FrankTseng168** [comment]
May I ask how to print the misjudgment case of yolo11 object detection in
the train, valid, and test sets?


Mohammed Yasin ***@***.***>於 2024年12月1日 週日，下午2:53寫道：

**Comment by Y-T-G** [comment]
Are you referring to confusion matrix?

**Comment by FrankTseng168** [comment]
Yes, but I want to find out the misclassified images!

Mohammed Yasin ***@***.***>於 2024年12月1日 週日，下午6:00寫道：

**Comment by FrankTseng168** [comment]
Could you let me know how to find out the misclassified images in confusion
matrix?

Mohammed Yasin ***@***.***>於 2024年12月1日 週日，下午6:00寫道：

**Comment by Y-T-G** [comment]
It's not supported in Ultralytics. You'll have to find it manually.

**Comment by FrankTseng168** [comment]
Ok, thanks.

Mohammed Yasin ***@***.***>於 2024年12月2日 週一，下午5:47寫道：

**Comment by glenn-jocher** [comment]
You're welcome! If you have further questions or need guidance on using specific Ultralytics tools, feel free to ask.
